backends/smt2/smtio.py

```diff
@@ -696,7 +704,9 @@ def check_sat(self):
                     if msg is not None:
                         print("%s waiting for solver (%s)" % (self.timestamp(), msg), flush=True)
 
-        result = self.read()
+        result = ""
```

Comment:
These changes are necessary because when solving with a `(maximize ...)` or `(minimize ...)` command, the solver outputs at least one line like:
```
((_ extract 8 1) s0) |-> 3
```
before the line with `sat` or `unsat`.

Reply:
I have now reverted this one change in master in c9555c9 because it broke other smt-based flows where yosys-smtbmc was not able to correctly identify situations where the solver produced unexpected output. Please create a new PR that keeps reading on the specific text patterns you need yosys-smtbmc to skip, not just anything that is not "sat" or "unsat".

Reply:
OK, no problem.  I think I will basically just have it skip lines with `|->` for now.

Can you give me an example failing `yosys-smtbmc` flow so I have something to test against?

Reply:
If you have SymbiYosys installed, try the following:
```
cd docs/examples/puzzles
sby -f primegen.sby primes_fail
```
The solver times out relatively quickly and returns `unknown`.

Reply:
OK, thanks.  Do you have any examples where the value returned is not `sat`, `unsat`, or `unknown`?  If it's always one of those three, I would prefer to keep this behavior (as it's necessary for `yosys-smtbmc` to work with optimizing exists-forall problems) and change the following line to:
```
        while result not in ["sat", "unsat", "unknown"]:
```

WDYT?

Reply:
See #1815